### PR TITLE
Group membership form

### PR DESCRIPTION
**Pull request by TechandEco**
This is a solution to be able to add new hires or team members (internal to a domain) to a Google Group via a Google Form.

Please review when you have a chance @asrivas and @erickoledadevrel

### CODE REVIEW

**Comment by erickoledadevrel** [comment]
What's the difference between this and #47? Might want to choose different folder names to make it more clear.

File: group-membership-form/README.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+For larger teams, you can optionally enable notficiations in the spreadsheet
+to receive an email everytime a new form submission occurs.
+
+## Technology highlights
+
+- The _onFormSubmit_ installable trigger is used to run the script upon
+  a user submitting an entry in a Google Form. To learn more visit
+  [this page](https://developers.google.com/apps-script/reference/script/form-trigger-builder#onFormSubmit())
+- Add members to a _Google Group for Business_ using the
+  [Admin Directory API](https://developers.google.com/apps-script/advanced/admin-sdk-directory)
```
Comment (asrivas) [inline group-membership-form/README.md]: is this officially called the Admin SDK?

File: group-membership-form/src/Code.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
+var ADDED_TO_GROUP_SUBJECT = 'Added to group';
+var ADDED_TO_GROUP_DOC_URL = 'https://docs.google.com/document/d/1-ajkkIP8gUWqMcnpXhkqwlM_2Y18USLdJ-pFZdDEZ70/edit?usp=sharing';
+
+var ALREADY_IN_GROUP_SUBJECT = 'Already in group';
+var ALREADY_IN_GROUP_DOC_URL = 'https://docs.google.com/document/d/11AO7vwk6179ohuxGO_NXSoDB0m_H5e-5XEtwiWRVNOM/edit?usp=sharing';
```
Comment (asrivas) [inline group-membership-form/src/Code.js]: Does this need a separate doc?

File: group-membership-form/src/Code.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
+var ADDED_TO_GROUP_SUBJECT = 'Added to group';
+var ADDED_TO_GROUP_DOC_URL = 'https://docs.google.com/document/d/1-ajkkIP8gUWqMcnpXhkqwlM_2Y18USLdJ-pFZdDEZ70/edit?usp=sharing';
```
[the remark this answers is not in the document]
Comment (asrivas) [reply group-membership-form/src/Code.js]: There's a specific Onboarding notes in the Docs Template Gallery we could use:
https://docs.google.com/document/u/0/?ftv=1

https://docs.google.com/document/d/1f4tE2Eyd4CgabQa-YJVQzF2SwpZJ-8aOgA2gesowxQ0/edit

File: group-membership-form/README.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
+---
+title: Offer a one stop shop onboarding experience
+description: Set new hires up for success with the resources they need
+labels:  Sheets, Forms, Groups for Business, Apps Script
```
[the remark this answers is not in the document]
Comment (TechandEco) [reply group-membership-form/README.md]: It's just groups which is weird

File: group-membership-form/src/Code.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (TechandEco) [reply group-membership-form/src/Code.js]: Good call, I updated the doc with that template! https://docs.google.com/document/d/1-ajkkIP8gUWqMcnpXhkqwlM_2Y18USLdJ-pFZdDEZ70/edit#

File: group-membership-form/README.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+[Twitter](https://twitter.com/TechandEco)!
+
+It’s hard to give incoming peers a consistently good experience when joining
+the team because of how error prone giving access to different resources can
+sometimes be. By using a Google Group, and sharing resources with that group's
+address, it makes it easy to help new hires get access to what they need to be
+productive in one step.
+
+If you have permissions to add users to a group, you
+can use this script to distribute the responsability to other members of your
```
[the remark this answers is not in the document]
Comment (TechandEco) [reply group-membership-form/README.md]: good catch!

File: group-membership-form/README.md (hunk 1 above)
Comment (TechandEco) [reply group-membership-form/README.md]: Yea if you look at the website it's called "Admin SDK Directory Service"

File: group-membership-form/src/Code.js (hunk 2 above)
Comment (TechandEco) [reply group-membership-form/src/Code.js]: Yes, for simplicity I have them separated. Based on the logic if a user is newly added they get a welcome letter and if they were already added they get an email that says they were already added. Didn't want it to get go silent.

**Pull request merged**